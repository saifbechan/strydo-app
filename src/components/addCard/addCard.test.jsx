import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import AddCard from './add-card.component';

const props = {
  columnId: 'abc-123',
  addCard: jest.fn()
};

it('renders correctly', () => {
  const { asFragment } = render(<AddCard {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it('calls addRow with the correct id on click', () => {
  const { getByTestId } = render(<AddCard {...props} />);

  fireEvent.click(getByTestId('add-card-handler'));

  expect(props.addCard).toHaveBeenCalledWith(props.columnId);
});
