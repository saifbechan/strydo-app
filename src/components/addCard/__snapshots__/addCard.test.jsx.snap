// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
}

<div
    class="c0"
    data-testid="add-card-handler"
  >
    + add card
  </div>
</DocumentFragment>
`;
